src/Command/CodeCheckerCommand.php

```diff
@@ -63,106 +45,69 @@ protected function configure()
-        //
-        // While this approach is logic and enabled to configure phpcs, we don't want that
-        // in this case, we just want to ensure phpcs knows about our standards,
-        // so we are going to force the installed_paths config here.
-        //
-        // And it needs need to do it BEFORE the runner init! (or it's lost)
-        //
-        // Note: the "moodle" one is not really needed, because it's autodetected normally,
-        // but the PHPCompatibility one is. There are some issues about version PHPCompatibility 10
-        // to stop requiring to be configured here, but that's future version. Revisit this when
-        // available.
-        //
-        // Note: the paths are relative to the base CodeSniffer directory, aka, the directory
-        // where "src" sits.
-        $runner->config->setConfigData('installed_paths', './../PHPCompatibility');
-        $runner->config->standards = [$this->standard]; // Also BEFORE init() or it's lost.
-
-        $runner->init();
-
-        $runner->config->parallel     = 1;
-        $runner->config->reports      = ['full' => null];
-        $runner->config->colors       = $output->isDecorated();
-        $runner->config->verbosity    = 0;
-        $runner->config->encoding     = 'utf-8';
-        $runner->config->showProgress = true;
-        $runner->config->showSources  = true;
-        $runner->config->interactive  = false;
-        $runner->config->cache        = false;
-        $runner->config->extensions   = ['php' => 'PHP'];
-        $runner->config->reportWidth  = 132;
-
-        $runner->config->files = $files;
-
-        $runner->config->setConfigData('testVersion', PHP_MAJOR_VERSION.'.'.PHP_MINOR_VERSION, true);
-
-        // Create the reporter to manage all the reports from the run.
-        $runner->reporter = new Reporter($runner->config);
-
-        // And build the file list to iterate over.
-        /** @var object[] $todo */
-        $todo         = new \PHP_CodeSniffer\Files\FileList($runner->config, $runner->ruleset);
-        $numFiles     = count($todo);
-        $numProcessed = 0;
-
-        foreach ($todo as $file) {
-            if ($file->ignored === false) {
-                try {
-                    $runner->processFile($file);
-                    ++$numProcessed;
-                    $runner->printProgress($file, $numFiles, $numProcessed);
-                } catch (\PHP_CodeSniffer\Exceptions\DeepExitException $e) {
-                    echo $e->getMessage();
-
-                    return $e->getCode();
-                } catch (\Exception $e) {
-                    $error = 'Problem during processing; checking has been aborted. The error message was: '.$e->getMessage();
-                    $file->addErrorOnLine($error, 1, 'Internal.Exception');
-                }
-                $file->cleanUp();
-            }
+        // Add the files to process.
+        foreach ($files as $file) {
+            $builder->add($file);
```

Comment:
I suppose we need to respect `($file->ignored === false)` before adding file?

Reply:
Yeah, from codesniffer source, it seems that the `$file->ignore` is used internally when the whole file is going to be ignored:

```
      /**
       * If TRUE, the entire file is being ignored.
       *
       * @var boolean
       */
      public $ignored = false;
```

And I think it's used when there are embed tags (`// phpcs ignore`...) and things like that. I don't see any place where it's being used to apply any list of excludes. It's just set when parsing the file.

So, conclusion... I think we are filtering (excluding) the files already (thirdtparty libraries, ignore paths and ignore names constants...) and weren't using any PHP_CodeSniffer filtering/excluding ability at all. And we have changed that here.

Tangentially related... the only point I was a little bit worried is if, now that we are executing the binary (from shell), maybe, the list of files could become too big (shell command limits) and I did some quick tests and accepted a fairly large list without problems. So finally assumed that it was ok to pass the files as part of the command (it's also possible to pass the list in a file, just in case we face problems with command max length).